operator/rediscontroller/pipeline.go

```diff
@@ -0,0 +1,226 @@
+package rediscontroller
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	exoscalesdk "github.com/exoscale/egoscale/v2"
+	"github.com/exoscale/egoscale/v2/api"
```

Comment:
I would probably introduce an alias to not confuse with kubernetes api
```suggestion
	exoscaleapi "github.com/exoscale/egoscale/v2/api"
```